src/Datadog.Trace/SpanContextPropagator.cs

```diff
@@ -166,6 +164,43 @@ public SpanContext Extract<T>(T carrier, Func<T, string, IEnumerable<string>> ge
             }
         }
 
+        private static TraceId? ParseTraceId<T>(T headers, string headerName)
+            where T : IHeadersCollection
+        {
+            var headerValues = headers.GetValues(headerName);
+            return ParseTraceId(headerValues, headerName);
+        }
+
+        private static TraceId? ParseTraceId<T>(T carrier, Func<T, string, IEnumerable<string>> getter, string headerName)
+        {
+            var headerValues = getter(carrier, headerName).ToList();
+            return ParseTraceId(headerValues, headerName);
+        }
+
+        private static TraceId? ParseTraceId(IEnumerable<string> headerValues, string headerName)
```

Comment:
It would be more orthogonal and readable to change the returning type to  `TraceId` and return `TraceId.Zero` instead of `null`. The usages would also be simplified.

Reply:
Done.

Reply:
Resolve